Python/ibmcloudsql/utilities.py

```diff
@@ -88,18 +89,25 @@ def logon(self, force=False):
             A new AIM token is created after 300 seconds.
             A token is valid for 3600 seconds
 
-            """
+        Raises
+        ---------
+        AttributeError
+           The exception is raised when the credential is incorrect.
+
+        """
         if self.logged_on and not force and (datetime.now() -
                                             self.last_logon).seconds < 300 and \
                                             ('authorization' in self.request_headers and
                                             'None' not in self.request_headers['authorization']):
-            return
+            return True
 
         ## TODO refactor construction to avoid calling private method
         boto3_session = self._session
         # ibm_boto3._get_default_session()
-        ro_credentials = boto3_session.get_credentials(
-        ).get_frozen_credentials()
+        try:
+            ro_credentials = boto3_session.get_credentials().get_frozen_credentials()
+        except ibm_botocore.exceptions.CredentialRetrievalError:
+            raise AttributeError("Login fails due to wrong credential - check the key")
```

Comment:
What's the reason for casting the exception into an `AttributeError` w/o details about `ibm_botocore.exceptions.CredentialRetrievalError`? it was designed to expose error info from `ibm_botocore.exceptions.CredentialRetrievalError` so I'm curious the motivation/use-case

Reply:
The purpose is to returns an exception, with an error message that a user can understand, and know what to fix, and if needed, can do try catching easily without having to load the `ibm_botocore` library which makes it looks more complicated than it should be.

Reply:
I think this can be separate PR itself (so that I will not block this PR review) since it's about UX on exception handling. IMHO, hiding details would benefit some user but will blind all users. Or you're saying the exception handling of `ibm_botocore` on authentication is not good enough for the users of this SDK

Reply:
Sure, I updated the code: (1) add error message, (2) maintain the error type.